country-dashboard/src/Component/Card.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useEffect, useState, Dispatch, SetStateAction } from "react";
+import axios from "axios";
+import { BASE_URL } from "./api";
+import { Box, Text, Image, Spinner, AspectRatio } from "@chakra-ui/react";
+
+interface CardsProps {
+  selectedCurrency: string | null;
+  selectedLanguage: string | null;
+  selectedRegion: string | null;
+  setSearchValue: Dispatch<SetStateAction<string>>;
+  searchValue: string;
+}
+
+export const Cards: React.FC<CardsProps> = ({
+  selectedCurrency,
+  selectedLanguage,
+  selectedRegion,
+  setSearchValue,
+  searchValue,
+}) => {
+  const [filteredCountries, setFilteredCountries] = useState<any[]>([]);
+  const [loading, setLoading] = useState<boolean>(false);
+  const [error, setError] = useState<string | null>(null);
+
+  const fetchData = async () => {
+    setLoading(true);
+    setError(null);
+    let url = BASE_URL + "all";
+    if (searchValue) {
+      url = BASE_URL + `name/${searchValue}`;
+    } else if (selectedCurrency) {
+      url = BASE_URL + `currency/${selectedCurrency}`;
+    } else if (selectedLanguage) {
+      url = BASE_URL + `lang/${selectedLanguage}`;
+    } else if (selectedRegion) {
+      url = BASE_URL + `region/${selectedRegion}`;
+    }
+
+    try {
+      const response = await axios.get(url);
+      const data = response.data;
+      setFilteredCountries(data);
+    } catch (error: any) {
+      console.error("Error fetching data:", error);
+      if (
+        axios.isAxiosError(error) &&
+        error.response &&
+        error.response.status === 404
+      ) {
+        setFilteredCountries([]);
+        setError("No country found");
+      } else {
+        setError("Error fetching data");
+      }
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    const timeoutId = setTimeout(fetchData, 300);
+
+    return () => {
+      clearTimeout(timeoutId);
+    };
+  }, [selectedCurrency, selectedLanguage, selectedRegion, searchValue]);
+
+  return (
+    <Box
+      display="grid"
+      gridTemplateColumns="repeat(auto-fill, minmax(250px, 1fr))"
+      gap={4}
+      p="7%"
+      pt="0px"
+      mt="0px"
+    >
+      {loading ? (
+        <Spinner size="lg" />
```

Comment:
Unfortunately not quite the loader that was in the design, we wanted a skeleton loader, -1